### PR TITLE
updated sync bn

### PR DESCRIPTION
**Pull request by ananyahjha93**
## What does this PR do?

Fixes # (issue)

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by ananyahjha93** [comment]
@williamFalcon not sure why the drone test is failing, the same test passes locally and was passing in the previous commit

**Comment by williamFalcon** [comment]
we don’t need that anymore since u added the test using ddp spawn no?
we don’t need the example anymore

**Comment by ananyahjha93** [comment]
ddp_spawn test is called using the example, as a script test

**Comment by Borda** [comment]
> we don’t need that anymore since u added the test using ddp spawn no?
> we don’t need the example anymore

the test is the example, maybe we shall put it to test as some other people crossing over would raise the same question...

File: tests/base/datamodules.py [hunk 1]
```diff
@@ -36,3 +38,47 @@ def val_dataloader(self):
...
+        self.batch_size = batch_size
+
+        # self.dims is returned when you call dm.size()
+        # Setting default dims here because we know them.
+        # Could optionally be assigned dynamically in dm.setup()
+        self.dims = (1, 28, 28)
+
+    def prepare_data(self):
+        # download only
+        MNIST(self.data_dir, train=True, download=True, normalize=(0.1307, 0.3081))
```
Comment (awaelchli) [inline tests/base/datamodules.py:60]: Just to double check, do we need TrialMNIST or MNIST?

File: tests/models/test_sync_batchnorm.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        if batch_idx == 2:
+            break
+
+    bn_outputs = [x.cuda() for x in bn_outputs]
+
+    # reset datamodule
+    # batch-size = 16 because 2 GPUs in DDP
+    dm = MNISTDataModule(batch_size=16, dist_sampler=True)
+    dm.prepare_data()
+    dm.setup(stage=None)
```
Comment (awaelchli) [inline tests/models/test_sync_batchnorm.py:84]: I thought the Trainer / LightningModule will call these automatically? @nateraw

File: tests/base/datamodules.py (hunk 1 above)
Comment (ananyahjha93) [reply tests/base/datamodules.py:60]: I did not want to change TrialMNIST structure for this PR as I had to include a DistributedSampler with shuffle

**Pull request merged**